src/Analysis/GWASResults/Utils/gwasWorkflowApi.js

```diff
@@ -2,16 +2,38 @@ import { headers } from '../../../configs';
 import { gwasWorkflowPath } from '../../../localconf';
 import { getPresignedUrl } from '../../AnalysisJob';
 
+export const getWorkflowDetails = async (
+  workflowName,
+  workflowUid,
+) => {
+  // query argo-wrapper endpoint to get the list of artifacts produced by the workflow:
+  const endPoint = `${gwasWorkflowPath}status/${workflowName}?uid=${workflowUid}`;
+  const response = await fetch(endPoint, { headers });
+  if (!response.ok) {
+    const message = `An error has occured: ${response.status}`;
+    throw new Error(message);
+  }
+  return response.json();
```

Comment:
duplicate code, consider creating a generic function to handle the api call and return json data